packetery/src/Packetery/Carrier/OptionsPage.php

```diff
@@ -0,0 +1,351 @@
+<?php
+/**
+ * Class OptionsPage
+ *
+ * @package Packetery\Options
+ */
+
+declare( strict_types=1 );
+
+namespace Packetery\Carrier;
+
+use Nette\Forms\Form;
+use Nette\Http\Request;
+use Packetery\FormFactory;
+
+/**
+ * Class OptionsPage
+ *
+ * @package Packetery\Options
+ */
+class OptionsPage {
+
+	/**
+	 * Latte_engine.
+	 *
+	 * @var \Latte\Engine Latte engine.
+	 */
+	private $latteEngine;
+
+	/**
+	 * Carrier repository.
+	 *
+	 * @var Repository Carrier repository.
+	 */
+	private $carrierRepository;
+
+	/**
+	 * Form factory.
+	 *
+	 * @var FormFactory Form factory.
+	 */
+	private $formFactory;
+
+	/**
+	 * Nette Request.
+	 *
+	 * @var Request Nette Request.
+	 */
+	private $httpRequest;
+
+	/**
+	 * Internal pickup points.
+	 *
+	 * @var string[] Internal pickup points.
+	 */
+	private $zpointCarriers;
+
+	/**
+	 * Plugin constructor.
+	 *
+	 * @param \Latte\Engine $latteEngine Latte_engine.
+	 * @param Repository    $carrierRepository Carrier repository.
+	 * @param FormFactory   $formFactory Form factory.
+	 * @param Request       $httpRequest Nette Request.
+	 */
+	public function __construct( \Latte\Engine $latteEngine, Repository $carrierRepository, FormFactory $formFactory, Request $httpRequest ) {
+		$this->latteEngine       = $latteEngine;
+		$this->carrierRepository = $carrierRepository;
+		$this->formFactory       = $formFactory;
+		$this->httpRequest       = $httpRequest;
+		$this->zpointCarriers    = [
+			'cz' => [
+				'id'   => 'zpointcz',
+				'name' => __( 'CZ Packeta pickup points', 'packetery' ),
+			],
+			'sk' => [
+				'id'   => 'zpointsk',
+				'name' => __( 'SK Packeta pickup points', 'packetery' ),
+			],
+			'hu' => [
+				'id'   => 'zpointhu',
+				'name' => __( 'HU Packeta pickup points', 'packetery' ),
+			],
+			'ro' => [
+				'id'   => 'zpointro',
+				'name' => __( 'RO Packeta pickup points', 'packetery' ),
+			],
+		];
+	}
+
+	/**
+	 * Registers WP callbacks.
+	 */
+	public function register(): void {
+		add_submenu_page(
+			'packeta-options',
+			__( 'Carrier settings', 'packetery' ),
+			__( 'Carrier settings', 'packetery' ),
+			'manage_options',
+			'packeta-country',
+			array(
+				$this,
+				'render',
+			),
+			10
+		);
+	}
+
+	/**
+	 * Creates settings form.
+	 *
+	 * @param array $carrierData Carrier data.
+	 *
+	 * @return \Nette\Forms\Form
+	 */
+	private function createForm( array $carrierData ): \Nette\Forms\Form {
+		$optionId = 'packetery_carrier_' . $carrierData['id'];
+		$form     = $this->formFactory->create( $optionId );
+		$form->setAction( $this->httpRequest->getUrl() );
+
+		$container = $form->addContainer( $optionId );
+
+		$container->addCheckbox(
+			'active',
+			__( 'Active carrier', 'packetery' )
+		);
+
+		$container->addText( 'name', __( 'Display name', 'packetery' ) )
+					->setRequired()
+					->addRule( $form::MIN_LENGTH, __( 'Carrier display name must have at least 2 characters!', 'packetery' ), 2 );
```

Comment:
Raději bych volil že se jedná o povinné pole (tedy musí být vyplněno),  neomezoval bych to na minimální počet znaků.